### PR TITLE
Create adsparc.js

### PR DESCRIPTION
**Pull request by yogeshadsparc**
Added Adsparc Adaptor JS Code

### CODE REVIEW

**Comment by protonate** [comment]
Thanks for submitting your adapter. Please add your additional changes to this branch and push the update for this PR. I'll close the additional PRs you've opened. This way we can assess the changes all at once.

File: src/adapters/adsparc.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+        _requestBid(bid);
+
+      }
+      // Only make one request per "nid"
+
+    }
+	
+    var getJSON = function(url) {
+      return new Promise(function(resolve, reject) {
```
Comment (snapwich) [inline src/adapters/adsparc.js]: You cannot use Promises from within Prebid.js.  Currently the build does not use babel-polyfill (which includes Promises) and Promises are not supported within Internet Explorer.

I'm also not sure on Prebid.js' support of Cross-origin AJAX.  Due to supporting IE9, you'd have to use the XDomainRequest object at the very least as regular CORs is not allowed within the XMLHttpRequest object in IE9.

If you would like to use JSONP for x-domain requests, there are helper functions here: https://github.com/prebid/Prebid.js/blob/master/src/adloader.js

**Comment by protonate** [comment]
Please also provide units tests for the adapter. We require > 80% code coverage. More info is [here](https://github.com/prebid/Prebid.js#unit-testing-with-karma).

File: src/adapters/adsparc.js (hunk 1 above)
Comment (yogeshadsparc) [inline src/adapters/adsparc.js]: Promises functions are removed.

**Comment by yogeshadsparc** [comment]
Unit Tests Are Added.

**Comment by yogeshadsparc** [comment]
Please let me know if you have any updates on this.

**Comment by protonate** [comment]
@yogeshadsparc can you please update with latest master and resolve build errors.

**Comment by yogeshadsparc** [comment]
Hey Hi, 
Someone please help me to add test cases.
My Code build is fine but giving error in test cases.
Error is here "Prebid Module getAdserverTargetingForAdUnitCodeStr should return targeting info as a string FAILED"

File: test/spec/unit/pbjs_api_spec.js [hunk 2]
```diff
@@ -105,7 +105,7 @@ describe('Unit: Prebid Module', function () {
...
+      var expected = 'foobar=300x250&hb_size=300x250&hb_pb=10.00&hb_adid=233bcbee889d46d&hb_bidder=appnexus&hb_size_triplelift=0x0&hb_pb_triplelift=10.00&hb_adid_triplelift=222bb26f9e8bd&hb_bidder_triplelift=triplelift&hb_size_appnexus=300x250&hb_pb_appnexus=10.00&hb_adid_appnexus=233bcbee889d46d&hb_bidder_appnexus=appnexus&hb_size_pagescience=300x250&hb_pb_pagescience=10.00&hb_adid_pagescience=25bedd4813632d7&hb_bidder_pagescienc=pagescience&hb_size_brightcom=300x250&hb_pb_brightcom=10.00&hb_adid_brightcom=26e0795ab963896&hb_bidder_brightcom=brightcom&hb_size_brealtime=300x250&hb_pb_brealtime=10.00&hb_adid_brealtime=275bd666f5a5a5d&hb_bidder_brealtime=brealtime&hb_size_pubmatic=300x250&hb_pb_pubmatic=10.00&hb_adid_pubmatic=28f4039c636b6a7&hb_bidder_pubmatic=pubmatic&hb_size_rubicon=300x600&hb_pb_rubicon=10.00&hb_adid_rubicon=29019e2ab586a5a&hb_bidder_rubicon=rubicon&hb_size_adsparc=300x250&hb_pb_adsparc=0.50&hb_adid_adsparc=3f88eb547&hb_bidder_adsparc=adsparc';
```
Comment (mkendall07) [inline test/spec/unit/pbjs_api_spec.js]: My guess is this is what's causing the error. Try rolling back the change in this file.

Comment (mkendall07) [inline test/spec/unit/pbjs_api_spec.js]: @yogeshadsparc 
Did you see this comment?

**Comment by mkendall07** [comment]
Closing. Please fix and resubmit

**Pull request closed without merging**